generator/generator.go

```diff
@@ -0,0 +1,149 @@
+	"fmt"
+	"os"
+	"strings"
+
+	_ "github.com/docker/machine/drivers/amazonec2"
+	//	_ "github.com/docker/machine/drivers/azure"
+	_ "github.com/docker/machine/drivers/digitalocean"
+	_ "github.com/docker/machine/drivers/exoscale"
+	//	_ "github.com/docker/machine/drivers/google"
+	_ "github.com/docker/machine/drivers/hyperv"
+	_ "github.com/docker/machine/drivers/packet"
+	_ "github.com/docker/machine/drivers/rackspace"
+	_ "github.com/docker/machine/drivers/softlayer"
+	_ "github.com/docker/machine/drivers/virtualbox"
+	_ "github.com/docker/machine/drivers/vmwarefusion"
+	_ "github.com/docker/machine/drivers/vmwarevcloudair"
+	_ "github.com/docker/machine/drivers/vmwarevsphere"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/codegangsta/cli"
+	"github.com/docker/machine/drivers"
+	"github.com/rancherio/go-machine-service/generator/helpers"
+)
+
+func main() {
+	err := generate()
+	if nil != err {
+		log.Fatal(err)
+	}
+}
+
+func generate() error {
+	resourceData, err := fetchData() // collects all data required to generate the files
+	if nil != err {
+		log.Errorf("error collecting driver data err=%v", err)
+		return err
+	}
+	err = helpers.GenerateAuthJsons(resourceData) // generate user_auth and project_auth jsons
+	if nil != err {
+		log.Errorf("error generating auth jsons err=%v", err)
+		return err
+	}
+	err = helpers.GenerateSpringContext(resourceData) // generate spring-docker-machine-api-context.xml
+	if nil != err {
+		log.Errorf("error generating spring context err=%v", err)
+		return err
+	}
+	err = helpers.GenerateFieldJsons(resourceData) // generate field information for each driver
+	if nil != err {
+		log.Errorf("error generating field jsons err=%v", err)
+		return err
+	}
+	err = helpers.GenerateMachineJson(resourceData) // generate schema/base/machine.json.d/machine-generated.json
+	if nil != err {
+		log.Errorf("error generating machine json err=%v", err)
+		return err
+	}
+	return err
+}
+
+func loadBlacklist() []string {
+	blacklist := os.Getenv("RANCHER_DRIVER_BLACKLIST")
+	if "" != blacklist {
+		return strings.Split(blacklist, ",")
+	}
+	// defaulting to this list for now
+	return []string{}
+}
+
+func fetchData() (*helpers.ResourceData, error) {
+	blacklist := loadBlacklist()
+	resourceMap := make(map[string]helpers.ResourceFields)
+	for _, driver := range drivers.GetDriverNames() {
+		resourceFieldStruct := make(helpers.ResourceFields)
+		resourceFieldMap := make(helpers.ResourceFieldConfigs)
+		resourceFieldStruct["resourceFields"] = resourceFieldMap
+		resourceMap[driver] = resourceFieldStruct
+		createFlags, err := drivers.GetCreateFlagsForDriver(driver)
+		if nil != err {
+			log.Errorf("error getting create flags for driver=%s, err=%v", driver, err)
+			return nil, err
+		}
+		for _, flagStruct := range createFlags {
+			var flagName, flagType string
+			var err error
+			switch flagStruct.(type) {
+			case cli.StringFlag:
+				flag := flagStruct.(cli.StringFlag)
+				flagName, err = getRancherName(flag.Name)
```

Comment:
This thing:
https://github.com/codegangsta/cli/blob/master/flag.go#L34-L39
but whatever, its fine as is.